controllers/extendeddaemonsetreplicaset/strategy/rollingupdate.go

```diff
@@ -125,6 +131,29 @@ func ManageDeployment(client client.Client, params *Parameters) (*Result, error)
 		result.Result.Requeue = true
 	}
 
+	// Remove canary labels from canary pods (if they exist)
+	// We keep retrying these operations only for the first X minutes after starting the rolling update to avoid Listing pods endlessly.
+	if time.Since(rollingUpdateStartTime) < cleanCanaryLabelsThreshold {
+		canaryPods := &corev1.PodList{}
+		listOptions := []runtimeclient.ListOption{
+			runtimeclient.MatchingLabels{
+				datadoghqv1alpha1.ExtendedDaemonSetReplicaSetCanaryLabelKey: datadoghqv1alpha1.ExtendedDaemonSetReplicaSetCanaryLabelValue,
+				datadoghqv1alpha1.ExtendedDaemonSetReplicaSetNameLabelKey:   params.Replicaset.GetName(),
+			},
+		}
+		if err = client.List(context.TODO(), canaryPods, listOptions...); err != nil {
+			params.Logger.Error(err, "Couldn't get canary pods")
```

Comment:
will this error continue reporting for `threshold` minutes after the label has been properly removed?

Reply:
no, we'll get an empty list and a nil error in this case